storm-server/src/main/java/org/apache/storm/DaemonConfig.java

```diff
@@ -800,6 +800,16 @@
     @NotNull
     public static final String SUPERVISOR_WORKER_START_TIMEOUT_SECS = "supervisor.worker.start.timeout.secs";
 
+    /**
+     * Maximum allowed time for how long a worker can go without heartbeating before the supervisor tries to restart the worker process.
+     * Upper limit for Config.TOPOLOGY_WORKER_TIMEOUT_SECS.
+     *
+     */
+    @IsInteger
+    @IsPositiveNumber
+    @NotNull
+    public static final String MAXIMUM_SUPERVISOR_WORKER_TIMEOUT_SECS = "supervisor.maximum.worker.timeout.secs";
```

Comment:
the variable name needs have same order or words just capitalized and _ as separator. So the va name would be SUPERVISOR_MAXIMUM_WORKER_TIMEOUT_SECS